dask_jobqueue/core.py

```diff
@@ -14,17 +19,119 @@ class JobQueueCluster(object):
+
     See Also
     --------
     PBSCluster
     SLURMCluster
     """
-    def __init__(self):
-        raise NotImplemented
+
+    _script_template ="""
+#!/bin/bash
+
+%(job_header)s
+
+%(worker_command)s 
+""".lstrip()
+
+    def __init__(self,
+                 name='dask-worker',
+                 threads=4,
+                 processes=6,
+                 memory='16GB',
+                 interface=None,
+                 death_timeout=60,
+                 local_directory=None,
+                 extra='',
+                 **kwargs
+                 ):
+        """
+        This initializer should be considered as Abstract, and never used directly.
+        :param name:
+        :param threads:
+        :param processes:
+        :param memory:
+        :param interface:
+        :param death_timeout:
+        :param local_directory:
+        :param extra:
+        :param kwargs:
+        """
+
+
+        if interface:
+            host = get_ip_interface(interface)
+            extra += ' --interface  %s ' % interface
+        else:
+            host = socket.gethostname()
+
+        self.cluster = LocalCluster(n_workers=0, ip=host, **kwargs)
+
+        self.jobs = dict()
+        self.n = 0
+        self._adaptive = None
+
+        self._command_template = "%s/dask-worker %s" % (dirname, self.scheduler.address)
+        if threads != None: self._command_template += " --nthreads %d" % threads
+        if processes != None: self._command_template += " --nprocs %d" % processes
+        if memory != None: self._command_template += " --memory-limit %s" % memory
+        if name != None:
+            self._command_template += " --name %s" % name
+            self._command_template += "-%(n)d" #Keep %(n) to be replaced later.
+        if death_timeout != None: self._command_template += " --death-timeout %s" % death_timeout
+        if local_directory != None: self._command_template += " --local-directory %s" % local_directory
+        if extra!= None: self._command_template += extra
 
     def job_script(self):
         self.n += 1
-        return self._template % toolz.merge(self.config, {'n': self.n})
+        return self._script_template % {'job_header': self.job_header,
+                                        'worker_command': self._command_template % {'n': self.n}
+                                        }
+
+    @property
+    def job_header(self):
+        """
+        Abstract attribute for the Job scheduler script header part.
+        :return: A string representation of the Job script header part.
+        """
+        raise NotImplementedError
+
+    @property
+    def submitcmd(self):
+        """
+        Abstract attribute for job scheduler submission command
+        :return:
+        """
+        raise NotImplementedError
+
+    @property
+    def cancelcmd(self):
+        """
+        Abstract attribute for job scheduler cancel command
+        :return:
+        """
+        raise NotImplementedError
```

Comment:
I'm in favor of removing these properties and just going with class attributes if possible